mobile-widgets/qmlmanager.cpp

```diff
@@ -301,10 +301,10 @@ void QMLManager::applicationStateChanged(Qt::ApplicationState state)
 	}
 	stateText.prepend("AppState changed to ");
 	stateText.append(" with ");
-	stateText.append((unsaved_changes() ? QLatin1String("") : QLatin1String("no ")) + QLatin1String("unsaved changes"));
+	stateText.append((!Command::isClean() ? QLatin1String("") : QLatin1String("no ")) + QLatin1String("unsaved changes"));
 	appendTextToLog(stateText);
 
-	if (state == Qt::ApplicationInactive && unsaved_changes()) {
+	if (state == Qt::ApplicationInactive && !Command::isClean()) {
```

Comment:
so this here, for example, might be incorrect - unless we carefully made sure that every path that uses the undo code also set our unsaved_changes() flag...